pyvista/core/utilities/geometric_sources.py

```diff
@@ -252,3 +295,208 @@ def output(self):
+        -------
+        sequence[float]
+            Direction vector in ``[x, y, z]``. Orientation vector of the
+            cylinder.
+        """
+        return self._direction
+
+    @direction.setter
+    def direction(self, direction: Sequence[float]):
+        """Set the direction in ``[x, y, z]``. Axis of the cylinder passes through this point.
+
+        Parameters
+        ----------
+        direction : sequence[float]
+            Direction vector in ``[x, y, z]``. Orientation vector of the
+            cylinder.
+        """
+        self._direction = direction
+
+    @property
+    def radius(self) -> bool:
+        """Get radius of the cylinder.
+
+        Returns
+        -------
+        float
+            Radius of the cylinder.
+        """
+        return self.GetRadius()
+
+    @radius.setter
+    def radius(self, radius: float):
+        """Set radius of the cylinder.
+
+        Parameters
+        ----------
+        radius : float
+            Radius of the cylinder.
+        """
+        self.SetRadius(radius)
+
+    @property
+    def height(self) -> float:
+        """Get the height of the cylinder.
+
+        Returns
+        -------
+        float
+            Height of the cylinder.
+        """
+        return self.GetHeight()
+
+    @height.setter
+    def height(self, height: float):
+        """Set the height of the cylinder.
+
+        Parameters
+        ----------
+        height : float
+            Height of the cylinder.
+        """
+        self.SetHeight(height)
+
+    @property
+    def resolution(self) -> int:
+        """Get number of points on the circular face of the cylinder.
+
+        Returns
+        -------
+        int
+            Number of points on the circular face of the cone.
+        """
+        return self.GetResolution()
+
+    @resolution.setter
+    def resolution(self, resolution: int):
+        """Set number of points on the circular face of the cone.
+
+        Parameters
+        ----------
+        resolution : int
+            Number of points on the circular face of the cone.
+        """
+        self.SetResolution(resolution)
+
+    @property
+    def capping(self) -> bool:
+        """Get cap cylinder ends with polygons.
+
+        Returns
+        -------
+        bool
+            Cap cylinder ends with polygons.
+        """
+        return self.GetCapping()
+
+    @capping.setter
+    def capping(self, capping: bool):
+        """Set cap cylinder ends with polygons.
+
+        Parameters
+        ----------
+        capping : bool, optional
+            Cap cylinder ends with polygons.
+        """
+        self.SetCapping(capping)
+
+    @property
+    def output(self):
+        """Get the output data object for a port on this algorithm.
+
+        Returns
+        -------
+        pyvista.PolyData
+            Cylinder surface.
+        """
+        self.Update()
+        output = wrap(self.GetOutput())
+        output.rotate_z(-90, inplace=True)
```

Comment:
Sorry I wasn't able to review this timely.  I have not tested this, but due to the rotation and translation in this method, will could we get different result in these two cases for `cyl_source = pv.CylinderSource()`, `cyl_source.output.plot()` and `pl.add_mesh(cyl_source)`.  I think the latter will not have this rotation and translation?

Reply:
We are always happy to review, no matter when. Please review whenever you like.
And very good point. This is a bug and I have created #4956 .